### PR TITLE
Release 0.5.2 to main

### PR DESCRIPTION
**Pull request by chiruu12**
## Summary

Promote **Release 0.5.2** to `main`, matching the prior release convention (`main` tracks the latest release).

`origin/dev` at **caa2e09** (Release 0.5.2 / #78) merges cleanly into `origin/main`. Contents since 0.5.1: store/Guard ML hardening (#72), inference fixes (#73), usability (#74), packaging (#75), shim coverage (#76), YAGNI removals (#77), and the 0.5.2 version bump (#78).

After merge: tag `v0.5.2` on `main` to trigger PyPI publish.

## Test plan
- [x] Local merge of `origin/dev` into `origin/main` has no conflicts
- [ ] PR CI green (if required)

---

> [!NOTE]
> **Medium Risk**
> Changes touch Guard ML init, model download/cache (user-facing upgrades), and dependency/install paths; misconfiguration or download failures are now stricter but better surfaced.
> 
> **Overview**
> **Release 0.5.2** bumps the SDK and changelog; functionally it tightens ML/runtime behavior, packaging, and public API/docs rather than adding new product surface.
> 
> **Guard & config:** Unknown `active_model` tiers now raise **`ConfigError`** with valid catalog names. **`judge_enabled`** and **`pipeline.judge_timeout` / `fail_closed`** are removed from Pydantic config (loader still **warns** if present in TOML); the optional judge runs only when passed as **`judge=`** to `Guard()`. **`Guard.init()`** docstring no longer claims auto-instrumentation. **`REVIEW`** pipeline outcomes log at **INFO** instead of WARNING.
> 
> **ML & model store:** Checkpoints must include weight files; corrupt manifests are ignored with warnings; downloads use atomic manifest writes and rollback so failed upgrades keep the previous checkpoint; **`list_status`** treats stale revisions as installed with upgrade available. Span load validates **`*-INJ`** labels, BIOES decode is safe without them, torch device forcing is validated, and **`ModelProvider` / `SpanInferenceModel`** loads are thread-safe. **`unplug-models download`** separates **`ImportError`** (install `[ml]`) from Hugging Face failures.
> 
> **Packaging & CI:** Version **0.5.2**; **`transformers`** constraint widened to **`>=4.44,<6`**. Published **`dev`** extra removed—pytest/ruff live in the **`dev` dependency group** (`uv sync --dev`). Python 3.13 CI skips heavy ML in the main matrix; new **`packaging-ml-wheels`** job runs **`uv pip compile --no-build`** for **`[ml]`**. Integrations live workflow installs **`--group dev`** instead of **`[dev]`**.
> 
> **Removals & docs:** Deletes unused **`providers/scrape.py`**, **`providers/content/server.py`**, **`optional/haystack.py`**. Docs clarify **`from unplug import …`** vs **`unplug.api.*`**. Broad new ML/unit tests (synthetic checkpoints, deprecation shims, store edge cases).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f38f1b2d8b93dd1063453dbdeba1ae52a81fe8ab. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: sdk/src/unplug/ml/store.py [hunk 1]
```diff
@@ -121,34 +166,57 @@ def ensure_tier(
...
+            if backup.exists():
+                shutil.rmtree(backup)
+            return dest
+        except Exception:
+            # Roll back to the last good checkpoint so a failed swap never
+            # strands the tier without a model (manifest still matches it).
+            if moved_to_backup and backup.exists():
+                if dest.exists():
+                    shutil.rmtree(dest)
+                os.replace(backup, dest)
```
[the remark this answers is not in the document]
Comment (chiruu12) [reply sdk/src/unplug/ml/store.py:216]: Addressed in #81 (merged to dev): https://github.com/UnplugAI/Unplug/pull/81

File: sdk/src/unplug/ml/store.py [hunk 2]
```diff
@@ -121,34 +166,57 @@ def ensure_tier(
...
-        )
-        ckpt = Path(local_dir)
-        if not self.is_valid_checkpoint(ckpt):
-            msg = (
-                f"Downloaded model at {ckpt} is missing config.json. "
-                f"Check repo {tier_entry.repo_id!r} on Hugging Face."
-            )
-            raise ConfigError(msg)
+        temp_dest = self.tier_dir(tier) / f".checkpoint-download-{os.getpid()}"
+        backup = self.tier_dir(tier) / f".checkpoint-{os.getpid()}.bak"
```
[the remark this answers is not in the document]
Comment (chiruu12) [reply sdk/src/unplug/ml/store.py:170]: Addressed in #81 (merged to dev): https://github.com/UnplugAI/Unplug/pull/81

File: sdk/src/unplug/ml/span_model.py [hunk 3]
```diff
@@ -68,7 +77,14 @@ def doc_threshold(self) -> float:
     def load(self) -> None:
         if self._model is not None:
             return
-        import torch
+        with self._load_lock:
+            if self._model is not None:
+                return
+            self._load_unlocked()
```
[the remark this answers is not in the document]
Comment (chiruu12) [reply sdk/src/unplug/ml/span_model.py:83]: Addressed in #81 (merged to dev): https://github.com/UnplugAI/Unplug/pull/81

**Comment by chiruu12** [comment]
Follow-up fixes merged in #81 → dev (`18dee378`). Model store staging/locking, manifest rollback, and span_model load/unload race.
